### PR TITLE
Error out when NativeLib has EventPipe enabled

### PR DESCRIPTION
**Pull request by LakshanF**
EventPipe does not work correctly with native AOT library (#89346). We will generate an error when `EventSourceSupport` property is set for native aot library.

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets [hunk 1]
```diff
@@ -50,6 +50,7 @@
       Text="RuntimeIdentifier is required for native compilation. Try running dotnet publish with the -r option value specified." />
     <Error Condition="'$(GeneratePackageOnBuild)' == 'true'" Text="GeneratePackageOnBuild is not supported for native compilation." />
     <Error Condition="'$(OutputType)' != 'Library' and '$(NativeLib)' != '' and '$(CustomNativeMain)' != 'true'" Text="NativeLib requires OutputType=Library." />
+    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventPipe library is not supported in NativeLib" />
```
Comment (agocke) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets:53]:
```suggestion
    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />
```

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets [hunk 2]
```diff
@@ -50,6 +50,7 @@
       Text="RuntimeIdentifier is required for native compilation. Try running dotnet publish with the -r option value specified." />
     <Error Condition="'$(GeneratePackageOnBuild)' == 'true'" Text="GeneratePackageOnBuild is not supported for native compilation." />
     <Error Condition="'$(OutputType)' != 'Library' and '$(NativeLib)' != '' and '$(CustomNativeMain)' != 'true'" Text="NativeLib requires OutputType=Library." />
+    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />
```
Comment (elinor-fung) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets:53]: Should we update the native AOT diagnostics doc you added to call out that it is not supported for native libraries?

Comment (LakshanF) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets:53]: I'm not sure on this one, as in this is too much of a corner case to call out on the official documentation and this error should be sufficient enough.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets (hunk 2 above)
Comment (jkotas) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets:53]:
```suggestion
    <!-- See https://github.com/dotnet/runtime/issues/89346 for details -->
    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />
```

**Pull request merged**